scanner/scanner.go

```diff
@@ -51,9 +51,13 @@ type ScannerOptions struct { // nolint:golint
 	// Number of fetched entries to buffer on their way to the callbacks
 	BufferSize int
 
-	// Log entry index to start fetching & matching at
+	// Log entry index to start fetching & matching at.
 	StartIndex int64
 
+	// Log entry index to finish fetching & matching at (non-inclusive).
+	// Scan up to the log's end if EndIndex == -1.
+	EndIndex int64
```

Comment:
What happens to all of the `scanner` users who implicitly leave this at the default value of `EndIndex: 0` ?

Reply:
Didn't realize those. How about making `EndIndex==0` a safe default meaning "end of log" instead of -1?